app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/CompareMenusBotViewModel.kt

```diff
@@ -68,7 +68,23 @@ class CompareMenusBotViewModel @Inject constructor(
                             eateries =
                             (listOfNotNull(firstEatery) + eateriesApiResponse.data.filter { it.name != firstEatery?.name }).filter { eatery ->
                                 filters.all { filter ->
-                                    filter.passesFilter(eatery, favorites, selected)
+                                    when (filter) {
+                                        is Filter.FromEatery -> filter.passesFilter(eatery)
+                                        is Filter.RequiresFavoriteEateries -> filter.passesFilter(
+                                            eatery,
+                                            favorites
+                                        )
+
+                                        is Filter.CustomFilter -> {
+                                            when (filter) {
+                                                Filter.CustomFilter.Selected -> selected.contains(
+                                                    eatery
+                                                )
+
+                                                Filter.CustomFilter.Today -> true
```

Comment:
Is the idea that this is just not a filter you'd see here? Maybe leave a comment if so :-)

Reply:
this is why I needed to refactor this lol